lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -81,35 +81,37 @@ sub engine_workit($) {
     # always set proper TAPDEV for os-autoinst when using tap network mode
     # ensure MAC addresses differ, tap devices may be bridged together
     # and allow MAC addresses for more than 256 workers (up to 65535)
-    if (($job->{'settings'}->{'NICTYPE'} // '') eq 'tap') {
-        $job->{'settings'}->{'TAPDEV'} = 'tap' . ($instance - 1);
-        $job->{'settings'}->{'NICMAC'} = sprintf("52:54:00:12:%02x:%02x", int($instance / 256), $instance % 256);
+    if (($job->{settings}->{NICTYPE} // '') eq 'tap') {
+        $job->{settings}->{TAPDEV} = 'tap' . ($instance - 1);
+        $job->{settings}->{NICMAC} = sprintf("52:54:00:12:%02x:%02x", int($instance / 256), $instance % 256);
     }
 
-    if (my $iso = $job->{'settings'}->{'ISO'}) {
-        $iso = join('/', ISO_DIR, $iso);
-        unless (-e $iso) {
-            warn "$iso does not exist!\n";
-            return;
+    for my $isokey (qw/ISO/, map { "ISO_$_" } (1 .. 9)) {
```

Comment:
Even if using more than 9 isos is not expected, what about
for my $isokey (grep { /^ISO/ } (keys $job->{settings}->{ISO}))